curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/653639d63a45a077333312c8.md

```diff
@@ -7,23 +7,27 @@ dashedName: step-32
 
 # --description--
 
-Before playing the song, you need to make sure it starts from the beginning. This can be achieved by the use of the `currentTime` property of the `audio` object. 
+Before playing the song, you need to make sure it starts from the beginning. This can be achieved by the use of the `currentTime` property on the `audio` object. 
 
-Add an `if` statement to check whether the `userData?.currentSong` is `null` or if `userData?.currentSong.id` is strictly not equal `song.id`. Inside `if` block, set the `currentTime` property of the `audio` object to `0`.
+Add an `if` statement to check whether the `userData?.currentSong` is falsy *OR* if `userData?.currentSong.id` is strictly not equal to `song.id`. This condition will check if no current song is playing or if the current song is different from the one that is about to be played.
 
+Inside your `if` block, set the `audio.currentTime` property to `0`.
 
 # --hints--
 
-You should create an `if` statement with the condition `userData?.currentSong === null || userData?.currentSong.id !== song.id`.
+You should create an `if` statement that checks whether `userData.currentSong` is either falsy *OR* `userData.currentSong.id` is not equal to `song.id`. Don't forget to use optional chaining.
```

Comment:
```suggestion
You should create an `if` statement that checks whether `userData?.currentSong` is either falsy *OR* `userData?.currentSong.id` is not equal to `song.id`. Don't forget to use optional chaining.
```

Reply:
uhm, no, this still passes without the required assignment:
![image](https://github.com/freeCodeCamp/freeCodeCamp/assets/26656284/d6554252-275d-47af-98a6-cad5a597473b)

Reply:
oh, wait, I had an outdated version locally. Sorry I didn't want to send that commit, let me fix that